### PR TITLE
Remove FlatFileBlockQuery

### PR DESCRIPTION
**Pull request by luckychess**
## What is this pull request?
PR removes FlatFileBlockQuery.
   
## Why do you implement it? Why do we need this pull request?
We don't use it anymore so we don't need it. Also an inheritance scheme looks strange with it: RedisFlatBlockQuery->FlatFileBlockQuery->BlockQuery; now it becomes just RedisFlatBlockQuery->BlockQuery.
  
## Details/Features
- Move methods which are still used from FlatFileBlockQuery to RedisFlatBlockQuery
- Remove FlatFileBlockQuery

### CODE REVIEW

File: irohad/ametsuchi/impl/redis_flat_block_query.hpp [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
+#include "ametsuchi/block_query.hpp"
 #include "ametsuchi/impl/flat_file/flat_file.hpp"
-#include "ametsuchi/impl/flat_file_block_query.hpp"
 
 #include "model/converters/json_block_factory.hpp"
 
 namespace iroha {
   namespace ametsuchi {
-    class RedisFlatBlockQuery : public FlatFileBlockQuery {
+    class RedisFlatBlockQuery : public BlockQuery {
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/redis_flat_block_query.hpp]: Add docs on this class

File: irohad/ametsuchi/impl/redis_flat_block_query.hpp [hunk 2]
```diff
@@ -19,16 +19,19 @@
...
 
 #include <boost/optional.hpp>
 
 namespace iroha {
   namespace ametsuchi {
-    class RedisFlatBlockQuery : public FlatFileBlockQuery {
+    /**
+     * Class which implements BlockQuery with a Redis backend.
+     */
+    class RedisFlatBlockQuery : public BlockQuery {
```
Comment (l4l) [inline irohad/ametsuchi/impl/redis_flat_block_query.hpp]: Maybe rename it to `RedisBlockQuery` as well?

File: irohad/ametsuchi/impl/redis_flat_block_query.cpp [hunk 3]
```diff
@@ -23,7 +23,55 @@ namespace iroha {
...
+
+    rxcpp::observable<model::Block> RedisFlatBlockQuery::getBlocksFrom(
+        uint32_t height) {
+      return getBlocks(height, block_store_.last_id());
+    }
+
+    rxcpp::observable<model::Block> RedisFlatBlockQuery::getTopBlocks(
+        uint32_t count) {
+      auto last_id = block_store_.last_id();
+      if (count > last_id) {
```
Comment (l4l) [inline irohad/ametsuchi/impl/redis_flat_block_query.cpp]: replace it with `std::min` plz

File: irohad/ametsuchi/impl/redis_flat_block_query.cpp [hunk 4]
```diff
@@ -23,7 +23,55 @@ namespace iroha {
...
     RedisFlatBlockQuery::RedisFlatBlockQuery(cpp_redis::redis_client &client,
                                              FlatFile &file_store)
-        : FlatFileBlockQuery(file_store), client_(client) {}
+        : block_store_(file_store), client_(client) {}
+
+    rxcpp::observable<model::Block> RedisFlatBlockQuery::getBlocks(
+        uint32_t height, uint32_t count) {
+      auto to = height + count;
+      auto last_id = block_store_.last_id();
+      if (to > last_id) {
```
Comment (l4l) [inline irohad/ametsuchi/impl/redis_flat_block_query.cpp]: replace it with `std::min` plz

File: test/module/irohad/ametsuchi/ametsuchi_test.cpp [hunk 5]
```diff
@@ -34,7 +34,7 @@
 #include "model/permissions.hpp"
 #include "module/irohad/ametsuchi/ametsuchi_fixture.hpp"
 
-#include "ametsuchi/impl/redis_flat_block_query.hpp"
+#include "ametsuchi/impl/redis_block_query.hpp"
```
Comment (l4l) [inline test/module/irohad/ametsuchi/ametsuchi_test.cpp]: Does this include is really required? If it changed and nothing changed in file that means yes

**Pull request merged**